test/test_response.py

```diff
@@ -606,6 +606,29 @@ def test_chunked_response_with_extensions(self):
         resp = HTTPResponse(r, preload_content=False, headers={'transfer-encoding': 'chunked'})
         assert stream == list(resp.stream())
 
+    def test_chunked_head_response(self):
+        r = httplib.HTTPResponse(MockSock, method='HEAD')
+        r.chunked = True
+        r.chunk_left = None
+        resp = HTTPResponse(MockChunkedEncodingResponse(''), preload_content=False,
+                            headers={'transfer-encoding': 'chunked'},
+                            original_response=r)
+        assert resp.chunked is True
+
+        class TrackRelease(object):
+            def __init__(self):
+                self.released = False
+
+        track_release = TrackRelease()
```

Comment:
Rather than use this inner class we can save ourselves some time and just store the calls in a list. Alternatively, you can use the mock module directly, which may be the best approach.